// Copyright (c) YugaByte, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software distributed under the License
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
// or implied.  See the License for the specific language governing permissions and limitations
// under the License.
//
syntax = "proto3";

package yb.docdb;

import "yb/common/common.proto";
import "yb/util/opid.proto";

option java_package = "org.yb.docdb";

message KeyValuePairPB {
  bytes key = 1;
  bytes value = 2;
}

// A set of key/value pairs to be written into RocksDB.
message KeyValueWriteBatchPB {
  repeated KeyValuePairPB kv_pairs = 1;
  TransactionMetadataPB transaction = 2;
  bool may_have_metadata = 3;
}

message ConsensusFrontierPB {
  OpIdPB op_id = 1;
  fixed64 hybrid_time = 2;
}
